src/librustdoc/html/render.rs

```diff
@@ -2194,6 +2194,7 @@ fn document(w: &mut fmt::Formatter, cx: &Context, item: &clean::Item) -> fmt::Re
         info!("Documenting {}", name);
     }
     document_stability(w, cx, item)?;
+    document_non_exhaustive(w, item)?;
```

Comment:
...though now i see why you did it this way. I commented in https://github.com/rust-lang/rust/issues/51860#issuecomment-402821544 that this marker could go into the "Fields" or "Variants" sections of the struct/enum pages, rather than at the top of the full docs.